src/main/java/org/apache/maven/shared/transfer/dependencies/resolve/internal/Maven30DependencyResolver.java

```diff
@@ -153,12 +154,12 @@
         final Class<?>[] argClasses =
             new Class<?>[] { org.apache.maven.model.Dependency.class, ArtifactTypeRegistry.class };
 
-         List<Dependency> aetherDependencies = new ArrayList<Dependency>( mavenDependencies.size() );
+         List<Dependency> aetherDependencies = mavenDependencies != null

```

Comment:
There's a serious bug in your proposed fix here which a test would probably reveal. It's helpful to write  a test for a bug before writing the fix to have a greater chance of catching these issues.

Reply:
Thanks for checking. Let me re-look at this case then.

Reply:
Right... I see `add` called in line 170 now.
Indeed I replaced `NPEx` here with `UnsupportedOperationException` just below. :disappointed:

Reply:
@elharo Please suggest UT change to expose that serious bug you mean.

Reply:
If you started with a test that assumed dependencies were null and failed with a NullPointerException, it likely would have caught this too.

Reply:
So I did - in the commit "Expose NPEx issue". Then the same test passes with my change.